crates/interpreter/src/interpreter.rs

```diff
@@ -252,3 +271,67 @@ impl Interpreter {
         }
     }
 }
+
+/// Enum representing the next action to be taken based on the interpreter's result.
+pub enum NextAction {
+    /// Indicates that the operation should continue.
+    Continue,
+    /// Indicates that the operation should revert.
+    Revert,
+    /// Indicates an external error occurred during the operation.
+    ExternalErr,
+    /// Indicates any other outcome not covered by the above variants.
+    Other,
+}
+/// Represents the outcome of a create operation in an interpreter.
+///
+/// This struct holds the result of the operation along with an optional address.
+/// It provides methods to determine the next action based on the result of the operation.
+pub struct CreateOutcome {
+    // The result of the interpreter operation.
+    result: InterpreterResult,
+    // An optional address associated with the create operation.
+    address: Option<Address>,
+}
+
+impl CreateOutcome {
+    pub fn new(result: InterpreterResult, address: Option<Address>) -> Self {
+        Self { result, address }
+    }
+    pub fn result(&self) -> &InterpreterResult {
+        &self.result
+    }
+    pub fn address(&self) -> Option<Address> {
+        self.address
+    }
+    /// Retrieves the data buffer based on the interpreter's result.
+    ///
+    /// If the result is `Revert`, `CallTooDeep`, or `OutOfFund`, it returns the associated output.
+    /// Otherwise, it returns an empty `Bytes` object.
```

Comment:
hmm, looks like we only need this in the interpreter?

I think hiding this here just obfuscates the check, so I'd recommend removing this because I don't think it's very useful